proposals/0025-dml-shader-correctness-test.md

```diff
@@ -0,0 +1,79 @@
+<!-- {% raw %} -->
+
+# DML Correctness Test
+
+* Proposal:
+  [0025-dml-shader-correctness-test](0025-dml-shader-correctness-test.md)
+* Author(s): [Sumit Agarwal](https://github.com/sumitsays)
+* Status: **Design In Progress**
+
+
+
+## Introduction
+
+Test suite to test the compute shader capability for the new hlsl compiler
+`Clang` is DirectML shaders. There will be 3 test stages:
+- Compilation test
+- Validation test
+- Correctness test   
+
+This document outlines the strategy to test the correctness of DirectML shaders
+against Clang. 
+
+## Background
+The relationship between DirectML shaders and DirectML API is *many-to-many*
+i.e. each DirectML shader is mapped to one or more DirectML public operator API
+and vice-versa. Aditionally, DirectML has TAEF based test to test each of their
+operator API via `DirectML.Test.OperatorTests.dll`.    
+Therefore we can leverage DirectML operator test infrastructure to test the
+correctness of DirectML shaders against Clang.
+
```

Comment:
Can we document what all the Shader stages do. Later you talk about concatenating and compressing but it isn't clear which stage that happens in. I'm assuming ShaderCompressor?